torch/_prims/__init__.py

```diff
@@ -0,0 +1,1000 @@
+    impl_aten=torch.le,
+    doc="",
+)
+
+lt = _make_elementwise_binary_prim(
+    impl_aten=torch.lt,
+    doc="",
+)
+
+max = _make_elementwise_binary_prim(
+    impl_aten=torch.maximum,
+    doc="",
+)
+
+min = _make_elementwise_binary_prim(
+    impl_aten=torch.minimum,
+    doc="",
+)
+
+mul = _make_elementwise_binary_prim(
+    impl_aten=torch.mul,
+    doc="",
+)
+
+ne = _make_elementwise_binary_prim(
+    impl_aten=torch.ne,
+    doc="",
+)
+
+nextafter = _make_elementwise_binary_prim(
+    impl_aten=torch.nextafter,
+    doc="",
+)
+
+pow = _make_elementwise_binary_prim(
+    impl_aten=torch.pow,
+    doc="",
+)
+
+rsqrt = _make_elementwise_binary_prim(
+    impl_aten=torch.rsqrt,
+    doc="",
+)
+
+shift_left = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_left_shift,
+    doc="",
+)
+
+shift_right_arithmetic = _make_elementwise_binary_prim(
+    impl_aten=torch.bitwise_right_shift,
+    doc="",
+)
+
+shift_right_logical = _not_impl
+
+sub = _make_elementwise_binary_prim(
+    impl_aten=torch.sub,
+    doc="",
+)
+
+#
+# View operations
+#
+def _broadcast_in_dim_meta(
+    a: TensorLike, shape: Sequence[int], broadcast_dimensions: Sequence[int]
+):
+    # Type checks
+    assert isinstance(a, TensorLike)
+    assert isinstance(shape, Sequence)
+    assert isinstance(broadcast_dimensions, Sequence)
+
+    # every dimension must be accounted for
+    assert a.ndim == len(broadcast_dimensions)
+
+    # broadcast shape must have weakly more dimensions
+    assert len(shape) >= a.ndim
+
+    # broadcast_dimensions must be an ascending sequence
+    # (no relative reordering of dims) of integers and
+    # each dimension must be within the new shape
+    def _greater_than_reduce(acc, x):
+        assert isinstance(x, int)
+        assert x > acc
+        assert x < len(shape)
+
+        return x
+
+    reduce(lambda acc, x: _greater_than_reduce(acc, x), broadcast_dimensions, -1)
+
+    # shape must be broadcastable to
+    for idx, new_idx in enumerate(broadcast_dimensions):
+        assert a.shape[idx] == 1 or a.shape[idx] == shape[new_idx]
+
+    new_strides = []
+    original_idx = 0
+    for idx in range(len(shape)):
+        if idx in broadcast_dimensions:
+            new_strides.append(a.stride()[original_idx])
+            original_idx = original_idx + 1
+        else:
+            new_strides.append(0)
+
+    return TensorMeta(a, shape=shape, strides=new_strides)
+
+
+def _broadcast_in_dim_aten(a, shape, broadcast_dimensions):
+    s = list(shape)
+    for broadcast_dimension in broadcast_dimensions:
+        s[broadcast_dimension] = -1
+
+    v = a
+    for idx, x in enumerate(s):
+        if x != -1:
+            v = v.unsqueeze(idx)
+
+    return v.expand(shape)
+
+
+_broadcast_in_dim_doc = """
+  Creates a view of t with the specified shape.
+
+  Allows adding dimensions of any length and broadcasting
+  dimensions of length one in t to any length.
+
+  The location of the broadcast dimensions must be specified
```

Comment:
THis seems the other way around? In lax docs (and in the implementation) "to which dimension in the target shape each dimension of the operand shape corresponds to"

Reply:
I believe we're saying the same thing I'm just not saying it very well